client/src/components/viewers/types.ts

```diff
@@ -0,0 +1,18 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { Translatable } from 'megashark-lib';
+
+export interface FileControlsDropdownItemContent {
+  id: number;
+  label: Translatable;
+  icon?: string;
+  children?: FileControlsDropdownGroup;
+  dismissPopover?: boolean;
+  dismissToParent?: boolean;
+  callback?: () => any;
```

Comment:
Shouldn't callback be mandatory? And since it's awaited, define it as `Promise<any>` instead.